Introducing gwdetchar-conlog

alurban opened the pull request:
This PR introduces a new command-line tool (`gwdetchar-conlog`) that identifies trends showing significant change between any two GPS times. Given a start time, stop time, IFO, and frametype, the script first identifies all channels that exist at both times, then reads frame data and prints channels whose values have changed.

`gwdetchar-conlog` also includes the following options:
* `--nproc`: number of parallel cores to use while reading data frames
* `--output`: optional filename to record output in, defaults to `changes.txt`
* `--channels`: optional list of regular expression patterns against which to require matching channel names, reducing the amount of frame reading and thus memory consumption and runtime

This fixes #272.

cc @duncanmmacleod, @jrsmith02, @andrew-lundgren, @tjma12

duncanmmacleod commented on bin/gwdetchar-conlog:
```diff
@@ -0,0 +1,107 @@
+#!/home/aurban/opt/gwdetchar-dev/bin/python
```
```suggestion
#!/usr/bin/env python
```

duncanmmacleod commented on bin/gwdetchar-conlog:
```diff
@@ -0,0 +1,107 @@
+times
+"""
+
+from __future__ import (division, print_function)
```
```suggestion
```

duncanmmacleod commented on bin/gwdetchar-conlog:
```diff
@@ -0,0 +1,107 @@
+
+import re
+import numpy
+import warnings
```
```suggestion
```

duncanmmacleod commented on bin/gwdetchar-conlog:
```diff
@@ -0,0 +1,107 @@
+    value2 = xoft2[0]
+    if value1 == value2:
+        continue
+    else:
```
This `else` can be removed:

```python
    if value1 == value2:
        continue
    messages.append('{} changes from {:.4g} to {:.4g}'.format(
            channel, value1, value2))
    print(messages[-1])

alurban commented:
@duncanmmacleod, thanks for taking a look, I've addressed your review comments. I've also tested the code and saved the output [**here**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/conlog/changes-1229317908-1229317928.csv) (there is also a log file [**here**](https://ldas-jobs.ligo-la.caltech.edu/~aurban/conlog/output.txt)).

jrsmith02 commented:
I love it. Likely we will tweak it as we use it. But LGTM
